File: CSharpTest/TypeUsageTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CSharpTest
{
    [TestClass]
    public class TypeUsageTest
    {
        [TestMethod]
        public void ObjectsTest()
        {
            var t = new MyLibrary.TypeUsageExample();
            t.PassByValue("PassByValue");
            Assert.AreEqual("PassByValue", t.ReturnByValue());
            t.PassByConstRef("PassByConstRef");
            Assert.AreEqual("PassByConstRef", t.ReturnByConstRef());
        }
    }
}

File: JavaTest/TypeUsageTest.java
import MyLibrary.*;
import org.junit.*;

import static org.junit.Assert.*;
import static org.junit.Assert.assertEquals;

public class TypeUsageTest extends TestBase {
    @Test
    public void ObjectsTest() {
        TypeUsageExample t = new TypeUsageExample();
        t.PassByValue("PassByValue");
        assertEquals("PassByValue", t.ReturnByValue());
        t.PassByConstRef("PassByConstRef");
        assertEquals("PassByConstRef", t.ReturnByConstRef());
    }
}
